rust/lance-index/src/scalar/inverted/wand.rs

```diff
@@ -546,7 +551,16 @@ impl<'a, S: Scorer> Wand<'a, S> {
                 let (Some(last), Some(next)) = (last, next) else {
                     return false;
                 };
-                max_relative_pos = Some(std::cmp::max(last, next) - slop);
+
+                let move_to = if last > next {
+                    last
+                } else {
+                    std::cmp::max(last + 1, next - slop)
+                };
+                max_relative_pos = match max_relative_pos {
```

Comment:
nit: I'm thinking of something like `max_relative_pos.unwrap_or(move_to).max(move_to)`

Reply:
Cool!